src/components/molecules/EditSpace/EditSpace.tsx

```diff
@@ -0,0 +1,221 @@
+
+import { RoomInput, upsertRoom } from "api/admin";
+
+import { RoomData_v2 } from "types/rooms";
+
+import { useUser } from "hooks/useUser";
+import { useVenueId } from "hooks/useVenueId";
+
+import ImageInput from "components/atoms/ImageInput";
+import { roomEditSchema } from "pages/Admin/Details/ValidationSchema";
+
+import "./EditSpace.scss";
+
+interface EditSpaceProps {
+  room: RoomData_v2;
+  updatedRoom: RoomData_v2;
+  roomIndex: number;
+  onBackPress: (roomIndex: number) => void;
+  onDelete?: () => void;
+  onEdit?: () => void;
+}
+export const EditSpace: React.FC<EditSpaceProps> = ({
+  room,
+  updatedRoom,
+  roomIndex,
+  onBackPress,
+  onDelete,
+  onEdit,
+}) => {
+  const [error, setError] = useState<string>("");
+
+  const { user } = useUser();
+
+  const venueId = useVenueId();
+
+  const { register, handleSubmit, setValue, watch, errors } = useForm({
+    reValidateMode: "onChange",
+    validationSchema: roomEditSchema,
+    defaultValues: {
+      title: room.title,
+      url: room.url,
+      isEnabled: room.isEnabled,
+      description: room.description,
+      template: room.template,
+      image_url: room.image_url,
+    },
+  });
+
+  const values = watch();
+
+  const handleImageChange = useCallback(
+    (val: string) => {
+      setValue("image_url", val, false);
+    },
+    [setValue]
+  );
+
+  const [{ loading: isUpdating }, updateRoom] = useAsyncFn(async () => {
+    if (!user || !venueId) return;
+    try {
+      const roomData = {
+        ...(room as RoomInput),
+        ...(updatedRoom as RoomInput),
+        ...values,
+      };
+
+      await upsertRoom(roomData, venueId, user, roomIndex);
+      onEdit && onEdit();
+    } catch (e) {
+      Bugsnag.notify(e, (event) => {
+        event.addMetadata("EditSpace::updateRoom", {
+          venueId: venueId,
+          roomIndex,
+        });
+      });
+    }
+  }, [onEdit, room, roomIndex, updatedRoom, user, values, venueId]);
+
+  const [{ loading: isDeleting }, deleteRoom] = useAsyncFn(async () => {
+    try {
+      await firebase.functions().httpsCallable("venue-deleteRoom")({
+        venueId,
+        room,
+      });
+      onDelete && onDelete();
+    } catch (error) {
+      setError("Can't delete room, please try again.");
+    }
+  }, [venueId, room, onDelete]);
+
+  const changeRoomState = () => {
+    setValue("isEnabled", !values.isEnabled);
+  };
+
+  return (
+    <Form onSubmit={handleSubmit(updateRoom)}>
+      <div className="EditSpace">
+        <div className="EditSpace__room">
+          <Form.Row>
+            <div className="room-edit-modal__input">
+              <Form.Label>Room type</Form.Label>
+              <Form.Control
+                type="text"
+                ref={register}
+                name="template"
+                value={room.template}
+                placeholder="Room name"
+                custom
+                disabled={true}
+              />
+            </div>
+          </Form.Row>
```

Comment:
Editing is not possible atm because the logic would add around 300-400 more lines of code and this PR will become ridiculously large. I will add it in a separate PR and for now this only displays the template of the created room.